File: FeedItForward/frontend/src/schemas/customerServiceSupportHistory.ts
import { Admin } from "./admin";
import { CSSMessage } from "./cssMessage";

export interface CustomerServiceSupportHistory {
  css_history_id: number;

  messages?: CSSMessage[];

  admin_id: number;
  admin: Admin;

  user_id: number;
}

export interface CustomerServiceSupportHistoryCreate {
  admin_id: number;
  user_id: number;
}

File: FeedItForward/backend/app/schemas/customer_service_support_history.py
from pydantic import BaseModel
from typing import Optional

from .css_message import CSSMessage
from .admin import Admin

class CustomerServiceSupportHistory(BaseModel):
    css_history_id: int

    messages: Optional[list[CSSMessage]] = []

    admin_id: int
    admin: Admin

    user_id: int

class CustomerServiceSupportHistoryCreate(BaseModel):
    admin_id: int
    user_id: int
